reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerRoutes.java

```diff
@@ -252,6 +252,15 @@ default HttpServerRoutes put(String path,
 		return route(HttpPredicate.put(path), handler);
 	}
 
+	/**
+	 * A generic route predicate that if matched not invoke already register I/O handler use
+	 * {@link HttpServerRoutes#route(Predicate, BiFunction)}.
+	 *
+	 * @param condition a predicate given each inbound request
+	 * @return this {@link HttpServerRoutes}
+	 */
+	HttpServerRoutes removeRoute(Predicate<? super HttpServerRequest> condition);
```

Comment:
how about to name that method as `removeIf`

Reply:
also, will it be sufficient to remove handler know just path? @ctlove0523 can you expand on your use case and how you wanna remove routes? 

Can it be enough to have `Predicate<? super HttpRouteHandlerMetad>` to find the required route?

Reply:
I'm thinking of something like 
```
HttpServerRoutes removeIf(Predicate<? super HttpRouteHandlerMetadata> predicate);
```

where the default implementation will utilize it as follows: 

```
@Override
	public HttpServerRoutes removeIf(Predicate<? super HttpRouteHandlerMetadata> predicate) {
		Objects.requireNonNull(predicate, "predicate");

		handlers.removeIf(predicate);

		return this;
	}
	```

Reply:
yes it has enough info to remove http route handler,i already change code like this.